app/models/degree_detail.rb

```diff
@@ -0,0 +1,22 @@
+class DegreeDetail
+  include ActiveModel::Model
+
+  attr_accessor :trainee, :degree_ids
+
+  validate :degrees_cannot_be_empty
+
+  def initialize(trainee)
+    @trainee = trainee
+    super(fields)
+  end
+
+  def fields
+    { degree_ids: trainee.degree_ids }
+  end
```

Comment:
Should be a private method right?

Reply:
Negative. It's called here `app/services/progress_service.rb:21`